buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py

```diff
@@ -0,0 +1,167 @@
+        behavior_file_path = Path(session_path) / f"{session_id}.behavior.mat"
+        behavior_mat = read_matlab_file(str(behavior_file_path))["behavior"]
+
+        # Add trials
+        events = behavior_mat["events"]
+        trial_interval_list = events["trialIntervals"]
+
+        data = []
+        for start_time, stop_time in trial_interval_list:
+            data.append(
+                dict(
+                    start_time=float(start_time),
+                    stop_time=float(stop_time),
+                )
+            )
+        [nwbfile.add_trial(**row) for row in sorted(data, key=lambda x: x["start_time"])]
+
+        trial_list = events["trials"]
+        direction_list = [trial.get("direction", "") for trial in trial_list]
+        trial_type_list = [trial.get("type", "") for trial in trial_list]
+
+        if not all([direction == "" for direction in direction_list]):
+            nwbfile.add_trial_column(name="direction", description="direction of the trial", data=direction_list)
+
+        if not all([trial_type == "" for trial_type in trial_type_list]):
+            nwbfile.add_trial_column(name="trial_type", description="type of trial", data=trial_type_list)
+
+        # Position
+        module_name = "Position"
+        module_description = "Contains behavioral data concerning position."
+        processing_module = get_module(nwbfile=nwbfile, name=module_name, description=module_description)
+
+        timestamps = np.array(behavior_mat["timestamps"])[..., 0]
+
+        position = behavior_mat["position"]
+        pos_data = [[x, y, z] for (x, y, z) in zip(position["x"], position["y"], position["y"])]
+        pos_data = np.array(pos_data)[..., 0]
+
+        units = behavior_mat.get("units", None)
+        if units == "m":
+            conversion = 1.0
+        else:
+            warnings.warn(f"Spatial units {units} not listed in meters; " "setting conversion to nan.")
+            conversion = np.nan
+
+        description = behavior_mat.get("description", "generic_position_tracking").replace("/", "-")
+        rotation_type = behavior_mat.get("rotationType", "non_specified")
+
+        pos_obj = Position(name=f"{description}")
+
+        spatial_series_object = SpatialSeries(
+            name=f"Position",
+            description=f"(x,y,z) coordinates tracking subject movement through",
+            data=H5DataIO(pos_data, compression="gzip"),
+            reference_frame="unknown",
+            conversion=conversion,
+            timestamps=timestamps,
+            resolution=np.nan,
+        )
+
+        pos_obj.add_spatial_series(spatial_series_object)
+
+        # Add error if available
+        errorPerMarker = behavior_mat.get("errorPerMarker", None)
+        if errorPerMarker:
+            error_data = np.array([error for error in errorPerMarker])[..., 0]
+
+            spatial_series_object = SpatialSeries(
+                name=f"Error per marker",
+                description=f"Error per marker",
```

Comment:
```suggestion
                description="Error per marker.",
```
I'd check for other unused f-strings as well.